### PR TITLE
Exchange repository test

### LINKED ISSUE #64: Add unit tests to ExchangeRepository
Add unit testing to ExchangeRepository. Something similar to the tests for other repositories we have should work. These tests should check all the logic contained in the repository, including validation logic.

### PR DESCRIPTION
**Pull request by moisses89**
Closes #64 
- test implemented logic
- test validation

### CODE REVIEW

File: src/domain/exchange/exchange.repository.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { JsonSchemaService } from '../../common/schemas/json-schema.service';
+import { ValidationErrorFactory } from '../errors/validation-error-factory';
+import { IExchangeApi } from '../interfaces/exchange-api.interface';
+import { ExchangeRepository } from './exchange.repository';
+import { ExchangeApi } from '../../datasources/exchange-api/exchange-api.service';
+import exchangeResultFactory from '../../domain/exchange/entities/__tests__/exchange.factory';
+import exchangeFiatCodesFactory from '../../domain/exchange/entities/__tests__/fiat-codes.factory';
+
+const validationErrorFactory = {
+  from: jest.fn().mockReturnValue(new HttpException('testErr', 500)),
```
Comment (fmrsabino) [inline src/domain/exchange/exchange.repository.spec.ts:11]: I'd say to mock any return value in the tests themselves. So the setup for the test is in one place.

File: src/domain/exchange/exchange.repository.spec.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+} as unknown as ValidationErrorFactory;
+
+const validationFunction = jest.fn();
+
+const jsonSchemaService = {
+  addSchema: jest.fn(),
+  compile: jest.fn().mockImplementation(() => validationFunction),
```
Comment (fmrsabino) [inline src/domain/exchange/exchange.repository.spec.ts:18]: Same as above ☝️

File: src/domain/exchange/exchange.repository.spec.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+} as unknown as JsonSchemaService;
+
+const exchangeApi = {
+  getFiatCodes: jest.fn().mockResolvedValue(ExchangeApi),
+  getExchangeResult: jest.fn().mockResolvedValue(ExchangeApi),
```
Comment (fmrsabino) [inline src/domain/exchange/exchange.repository.spec.ts:23]: Same as above ☝️

File: src/domain/exchange/exchange.repository.spec.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  let exchangeRepository: ExchangeRepository;
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    exchangeRepository = new ExchangeRepository(
+      mockExchangeApi,
+      mockValidationErrorFactory,
+      mockJsonSchemaService,
+    );
+    validationFunction.mockImplementation(() => true);
```
Comment (fmrsabino) [inline src/domain/exchange/exchange.repository.spec.ts:40]: Same as above ☝️

File: src/domain/exchange/exchange.repository.spec.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+  it('should return the Fiat Codes', async () => {
+    const exchangeFiatCodes = exchangeFiatCodesFactory(true, {
+      USD: 'dollar',
+      AED: 'dirham',
+    });
+    mockExchangeApi.getFiatCodes.mockResolvedValue(exchangeFiatCodes);
+
+    const result = await exchangeRepository.getFiatCodes();
+    expect(Object.keys(exchangeFiatCodes.symbols)).toStrictEqual(result);
```
Comment (fmrsabino) [inline src/domain/exchange/exchange.repository.spec.ts:51]: nitpick:

```
// setup

// test

// assert
```

File: src/domain/exchange/exchange.repository.spec.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+  it('Should return convert rate', async () => {
+    const exchangeResult = exchangeResultFactory(true, {
+      BIG: 10,
+      LIT: 1,
+    });
+    mockExchangeApi.getExchangeResult.mockResolvedValue(exchangeResult);
+
+    const result = await exchangeRepository.convertRates('LIT', 'BIG');
+    expect(0.1).toBe(result);
```
Comment (fmrsabino) [inline src/domain/exchange/exchange.repository.spec.ts:68]: nitpick:

```
// setup

// test

// assert
```

**Review by fmrsabino — CHANGES_REQUESTED** [review]

File: src/domain/exchange/exchange.repository.spec.ts (hunk 2 above)
Comment (moisses89) [reply src/domain/exchange/exchange.repository.spec.ts:17]: I couldn't find any clean way to move this to the test. I mean `validationfunction` must be included before instantiate the exchangeRepository.

**Pull request merged**